### ISSUE TITLE
Make fetch async

### ISSUE THREAD

**Issue by PerWiklander** [ochrons/diode#8]
I sometimes get this message in the console when using `PotMap`:

```
Warning: DiodeWrapper is accessing isMounted inside its render() function.
render() should be a pure function of props and state. It should never
access something that requires stale data from the previous render,
such as refs. Move this logic to componentDidMount and componentDidUpdate instead.
```

It looks like the problem is that the `mapHandler` in `AsyncActionRetriable` wants to set the value of the `Pot` to `Pending` right away. If this happens while rendering, React gets cranky.

**Comment by ochrons** [comment]
This is due to `PotMap` calling fetcher synchronously, which updates the model and causes a chain of calls that result in a call to `isMounted` in the `ReactConnector`. I'll update `PotMap` to always call fetcher in an async fashion to avoid this (and other potential related problems).

**Comment by ochrons** [comment]
Fixed in ece23b64e3e57ec1d6da963702885c6e3e3eeb92
